File: 백준/Silver/1417. 국회의원 선거/국회의원 선거.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int dasom = Integer.parseInt(br.readLine());
        int[] candi = new int[n-1];
        int res = 0;
        for(int i=0;i<n-1;i++){
            candi[i] = Integer.parseInt(br.readLine());
        }
        if(n==1){
            System.out.println(0);
            return;
        }

        while(true){
            int maxIdx = 0;
            for(int i=1;i<candi.length;i++){
                if(candi[i]>candi[maxIdx]){
                    maxIdx = i;
                }
            }
            if(candi[maxIdx]<dasom)break;
            candi[maxIdx]-=1;
            dasom +=1;
            res+=1;
        }
        System.out.println(res);
    }
}

File: 백준/Silver/1417. 국회의원 선거/국회의원 선거.py
import sys
input = sys.stdin.readline
n = int(input())
dasom = int(input())
candi = [int(input()) for _ in range(n-1)]
res= 0
if not candi:
    print(0)
    exit()

while max(candi) >=dasom:
    candi.sort(reverse=True)
    candi[0]-=1
    dasom +=1
    res +=1
print(res)
